### DISCUSSION TITLE
同一张图片使用在线模式识别的结果和使用本地识别的结果差异很大，原因是什么？

### DISCUSSION THREAD

**Discussion by Dkaaaaa** [RapidAI/TableStructureRec#93] [Q&A] [answered]
您好，我在使用TableStructureRec的时候出现了一下问题：
这是我的表格图片：![imgf000242_0001-rotated](https://github.com/user-attachments/assets/4c6444c4-e2b8-4b89-b017-db9706d7f32e)
这是在线模式的参数：
![image](https://github.com/user-attachments/assets/be80611c-582a-4639-b2aa-de49e893a180)
这是在线模式的识别结果：
![image](https://github.com/user-attachments/assets/34241ed2-eeca-4c60-bf06-092319f7055f)

我本地代码所使用的参数与在线模式的一致，但是结果却差的很远，这是我的本地运行结果：
![image](https://github.com/user-attachments/assets/f2e67eee-a234-4c45-a27f-bddee9da4289)

我算了一下这张图片的拉普拉斯指数为2731，想知道是什么原因导致这两种方式获得的结果的差异？

**Comment by Joker1212** [comment]
> 您好，我在使用TableStructureRec的时候出现了一下问题： 这是我的表格图片：![imgf000242_0001-rotated](https://private-user-images.githubusercontent.com/109192357/394170866-4c6444c4-e2b8-4b89-b017-db9706d7f32e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTc2MTQsIm5iZiI6MTczMzgxNzMxNCwicGF0aCI6Ii8xMDkxOTIzNTcvMzk0MTcwODY2LTRjNjQ0NGM0LWUyYjgtNGI4OS1iMDE3LWRiOTcwNmQ3ZjMyZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQwNzU1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDA5NzQyYWNmMzZlMmYzMjg3MmEyNGUzNjc5MmVlNzQyZWM4NGNhM2E0Mzc2ZGExZjMwNTBjMTJiMzE2MzkzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7kV7BABZlJlyYDXi_m0aM7xjMoEAmT5tp0bgkYm1JNo) 这是在线模式的参数： ![image](https://private-user-images.githubusercontent.com/109192357/394171138-be80611c-582a-4639-b2aa-de49e893a180.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTc2MTQsIm5iZiI6MTczMzgxNzMxNCwicGF0aCI6Ii8xMDkxOTIzNTcvMzk0MTcxMTM4LWJlODA2MTFjLTU4MmEtNDYzOS1iMmFhLWRlNDllODkzYTE4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQwNzU1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZTE1ZWVjM2UwNjNiN2QxMTY4NTAwMTE5MTQxNjU0NjM0MWUxNDc2MjZhNmNjZDIzMzBhZGY0MGI2NTg3ZjE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.tBmkMwnaAFpv6sV18fcmq4vq4XmME03UuYoru24z_VQ) 这是在线模式的识别结果： ![image](https://private-user-images.githubusercontent.com/109192357/394171433-34241ed2-eeca-4c60-bf06-092319f7055f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTc2MTQsIm5iZiI6MTczMzgxNzMxNCwicGF0aCI6Ii8xMDkxOTIzNTcvMzk0MTcxNDMzLTM0MjQxZWQyLWVlY2EtNGM2MC1iZjA2LTA5MjMxOWY3MDU1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQwNzU1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMDdhMDM5M2U4OTNkNzgzZDdmOWY0ZmE2ZWI4NGQyZmM3MjE1ZjljMDBhNzcyMzYzZGQyOWNlZGQxODE4YTJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.80BbAsP7gk6S3QG502uuXGJapURQp-BIev41UEqr-as)
> 
> 我本地代码所使用的参数与在线模式的一致，但是结果却差的很远，这是我的本地运行结果： ![image](https://private-user-images.githubusercontent.com/109192357/394171740-f2e67eee-a234-4c45-a27f-bddee9da4289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTc2MTQsIm5iZiI6MTczMzgxNzMxNCwicGF0aCI6Ii8xMDkxOTIzNTcvMzk0MTcxNzQwLWYyZTY3ZWVlLWEyMzQtNGM0NS1hMjdmLWJkZGVlOWRhNDI4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQwNzU1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMmNiYTY1MTk1YzJkYWM1NWYyZTBlNjEyMDc3YTlkNzIyMjBmOWMyYTJiYWEwOThiMjY5NzkzOTFkYzhmYjkwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yzQItIvYSp1QrdgXYs1c8rE-VX-MZ0q-SP5_JKeIgkw)
> 
> 我算了一下这张图片的拉普拉斯指数为2731，想知道是什么原因导致这两种方式获得的结果的差异？

emmm,会不会是表格分类出问题了，试试直接使用有线表格来做呢？因为表格onnx的执行其实不同平台执行结果并不是完全一致，可能给分类为无线表格了,也可以切换不同的表格分类模型来试试，有一个trick是如果图片很大，可以适当压缩

**Reply by Dkaaaaa** [reply]
刚刚试了下，果然限制为有线表格就结果一致了。
您指的图片很大是指图片的尺寸（分辨率）吗？然后我也有点好奇，如果图片很大会造成什么影响？因为我看rapidocr里面是有设置最大图片尺寸的参数的。然后就是模型在读入照片的时候，是不是会对图片尺寸进行重新调整呢？因为我调大过图片的尺寸，发现对最终的识别结果没什么影响。
还有就是有一些单元格会被分成两个单元格，类似于这种情况，是可以设置col_threshold和col_threshold来避免吗，还是说也可以通过缩小图片尺寸来避免？
![image](https://github.com/user-attachments/assets/cbc7e36d-21cc-4e82-860e-353c3f030e03)

**Reply by Joker1212** [reply]
理论上，有参数能解决这个合并，但是得精调了，一般来说把分辨率减小也可以避免，原理其实我也不太确定，像是这个unet模型的一个特点

**Reply by Dkaaaaa** [reply]
想知道您说的参数可以在哪里了解到

**Reply by Joker1212** [reply]
建议是，缩小一下分辨率，如果你有很多表格希望自动化，换一下表格分类模型(yolo,yolox,q)，精调的参数因为有点难描述而且比较多，我没有写在readme，之前的某个版本其实写了

**Reply by Joker1212** [reply, marked as answer]
我刚刚试了，等比缩小到2004 * 1446就行了，应该还是阈值判断都是给定的比如10,20这样的像素值，当分辨率大的时候，条件容易满足，所以就更容易延伸水平线和竖线

**Reply by Joker1212** [reply]
wired_table_rec = WiredTableRecognition()
html, elasp, polygons, logic_points, ocr_res = wired_table_rec(
    img_path,
    version="v2", #默认使用v2线框模型，切换阿里读光模型可改为v1
    morph_close=True, # 是否进行形态学操作,辅助找到更多线框,默认为True
    more_h_lines=True, # 是否基于线框检测结果进行更多水平线检查，辅助找到更小线框, 默认为True
    h_lines_threshold = 100, # 必须开启more_h_lines, 连接横线检测像素阈值，小于该值会生成新横线，默认为100
    more_v_lines=True, # 是否基于线框检测结果进行更多垂直线检查，辅助找到更小线框, 默认为True
    v_lines_threshold = 15, # 必须开启more_v_lines, 连接竖线检测像素阈值，小于该值会生成新竖线，默认为15
    extend_line=True, # 是否基于线框检测结果进行线段延长，辅助找到更多线框, 默认为True
    need_ocr=True, # 是否进行OCR识别, 默认为True
    rec_again=True,# 是否针对未识别到文字的表格框,进行单独截取再识别,默认为True
)
lineless_table_rec = LinelessTableRecognition()
html, elasp, polygons, logic_points, ocr_res = lineless_table_rec(
    need_ocr=True, # 是否进行OCR识别, 默认为True
    rec_again=True,# 是否针对未识别到文字的表格框,进行单独截取再识别,默认为True
)

**Reply by Dkaaaaa** [reply]
好滴，帮大忙了，谢谢！
